### PR TITLE
ADD options=value mode for NGSIv2 entity queries

### PR DESCRIPTION
**Pull request by fgalan**
Implements #1049 

@kzangeli 
@crbrox

### CODE REVIEW

File: src/lib/apiTypesV2/Entity.cpp [hunk 1]
```diff
@@ -81,27 +81,39 @@ std::string Entity::render(ConnectionInfo* ciP, RequestType requestType, bool co
...
     {
+      std::string out = "[";
+      if (attributeVector.size() != 0)
+      {
+        out += attributeVector.toJson(true, false, renderMode, ciP->uriParam["attrs"]);
+      }
+      out += "]";
+      return out;
+    }
+    else
```
Comment (kzangeli) [inline src/lib/apiTypesV2/Entity.cpp]: As you return after 'renderMode == "values"', the else part can be removed, I mean, the content of the else part can be outside the if-else and be less indented.

Comment (fgalan) [inline src/lib/apiTypesV2/Entity.cpp]: Fixed in 1a99b1b7525c63f4b01aba431a07e16a60703fa1 (actually the else is needed, as there is common code).

File: test/functionalTest/cases/1049_values_mode/values_mode.test [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# Note that we need to use curl directly in steps 2-5 to avoid
+# the field order done by the pretty printer
+#
```
Comment (kzangeli) [inline test/functionalTest/cases/1049_values_mode/values_mode.test]: erroneous cut 'n paste?

File: test/functionalTest/cases/1049_values_mode/values_mode_entities.test [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# Note that we need to use curl directly in steps 2-5 to avoid
+# the field order done by the pretty printer
```
Comment (kzangeli) [inline test/functionalTest/cases/1049_values_mode/values_mode_entities.test]: same same

Comment (fgalan) [inline test/functionalTest/cases/1049_values_mode/values_mode_entities.test]: Fixed in 79018de

File: test/functionalTest/cases/1049_values_mode/values_mode.test (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/cases/1049_values_mode/values_mode.test]: Fixed in 79018de

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**